### ISSUE TITLE
no-named-as-default case sensitivity option

### ISSUE THREAD

**Issue by perrin4869** [benmosher/eslint-plugin-import#468]
Would it be possible to add an option to the `no-named-as-default` to ignore case sensitivity? In my project all my files are named in lower case. However, when I import a class I will get an error, for example:

```js
import Login from '../../client/js/components/login.jsx';
```

results in the following error:

```
error  Using exported name 'Login' as identifier for default export      import/no-named-as-default
```

Would it be possible to add an option to the rule, to allow for case insensitive matching? Thanks!

**Comment by benmosher** [comment]
Can you post the corresponding export from `login.jsx`?

This rule warns if `login.jsx` exports both `default` and `Login`. It's not based on the filename, but the actual exports of the file itself (which are case-sensitive).

If you intend to import the `default` export with a name that is also exported (in this case, `Login`), you will probably want to disable `no-named-as-default` as it is meant to warn in exactly this case.

**Comment by perrin4869** [comment]
This is what I export:

```js
import { connect } from 'react-redux';

export const Login = props => (<div>login</div>);
export default connect(mapStateToProps)(Login);
```

This is common practice used for testing components without having them wrapped by `react-redux`, documented in the official redux docs... I guess there is no better way around it, rather than disabling the rule?

**Comment by benmosher** [comment]
Nope, that's pretty much the situation that this rule intends to protect you from. Only alternative I can think of would be to `//eslint-disable-line import/no-named-as-default`, but that's gnarly, IMO.

**Comment by perrin4869** [comment]
got it, thanks!

**Comment by brendonco** [comment]
Darn disable eslint
